src/library/dao/directorydao.cpp

```diff
@@ -62,27 +62,25 @@ DirectoryDAO::AddResult DirectoryDAO::addDirectory(
     const auto newCanonicalLocation = newDir.canonicalLocation();
     DEBUG_ASSERT(!newCanonicalLocation.isEmpty());
     QList<mixxx::FileInfo> obsoleteChildDirs;
-    for (auto&& oldDir : loadAllDirectories()) {
-        if (!oldDir.exists() || !oldDir.isDir()) {
-            // Abort to prevent inconsistencies in the database
-            kLogger.warning()
-                    << "Aborting to add"
-                    << newDir
-                    << ": Loaded directory"
-                    << oldDir
-                    << "does not exist or is inaccessible";
-            return AddResult::InvalidOrMissingDirectory;
-        }
+    // Ignore invalid or missing directories in order to aloow adding new dirs
+    // while the list contains e.g. currently unmounted removable drives.
+    // Worst that can happen is that we have orphan tracks in the database that
+    // would be moved to missing after the rescan (which is required anyway after
+    // having added a new dir).
+    for (auto&& oldDir : loadAllDirectories(true)) {
```

Comment:
Alternatively, use the current implemantation but don't return if the old dir is missing. Instead, do the QString equivalent of `mixxx::FileInfo::isRootSubCanonicalLocation(new, old)` in that case.
This requires refactirng here, i.e. use `DirectoryDAO::getRootDirStrings()` added in #12878 and construct the mixxx::FileInfo here.